drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,217 @@
+};
+
+/* BELLBOARD IRQ list for each group */
+#define GRP_IRQ_GET_OR_ZERO(grp)                                                                   \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(0, grp), (DT_INST_IRQ_BY_NAME(0, grp, irq)), (0U))
+
+static const uint8_t irqs[BELLBOARD_IRQ_COUNT] = {
+	GRP_IRQ_GET_OR_ZERO(bellboard0),
+	GRP_IRQ_GET_OR_ZERO(bellboard1),
+	GRP_IRQ_GET_OR_ZERO(bellboard2),
+	GRP_IRQ_GET_OR_ZERO(bellboard3),
+};
+
+/* BELLBOARD IRQ mappings */
+#define GET_IRQ_MAPPING_OR_ZERO(idx)                                                               \
+	COND_CODE_1(DT_INST_PROP_HAS_IDX(0, nordic_interrupt_mapping, idx),                        \
+		    (DT_INST_PROP_BY_IDX(0, nordic_interrupt_mapping, idx)), (0U))
+
+static const uint32_t irq_mappings[BELLBOARD_IRQ_COUNT] = {
+	GET_IRQ_MAPPING_OR_ZERO(0),
+	GET_IRQ_MAPPING_OR_ZERO(2),
+	GET_IRQ_MAPPING_OR_ZERO(4),
+	GET_IRQ_MAPPING_OR_ZERO(6),
+};
+
+/* BELLBOARD instance */
+static NRF_BELLBOARD_Type *bellboard = (NRF_BELLBOARD_Type *)DT_INST_REG_ADDR(0);
+
+BUILD_ASSERT(DT_INST_PROP_LEN(0, nordic_interrupt_mapping) == 2 * DT_NUM_IRQS(DT_DRV_INST(0)),
+	     "# interrupt mappings != # interrupts");
+
+static void bellboard_local_isr(const void *parameter)
+{
+	uint8_t grp_idx = (uintptr_t)parameter;
+	struct mbox_bellboard_local_group *grp = grps[grp_idx];
+	uint32_t int_pend;
+
+	int_pend = nrfy_bellboard_int_pending_get(bellboard, grp_idx);
+	(void)nrfy_bellboard_events_process(bellboard, int_pend);
+
+	for (uint8_t i = 0U; i < NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT; i++) {
+		if ((int_pend & BIT(i)) != 0U) {
+			if (grp->cb[i] != NULL) {
+				grp->cb[i](DEVICE_DT_INST_GET(0), i, grp->user_data[i], NULL);
+			}
+		}
+	}
+}
+
+static int bellboard_local_grp_get(uint32_t id, uint8_t *grp)
+{
+	if (id >= NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT) {
+		return -EINVAL;
+	}
+
+	for (uint8_t i = 0U; i < ARRAY_SIZE(irq_mappings); i++) {
+		if ((irq_mappings[i] & BIT(id)) != 0U) {
+			*grp = i;
+			return 0;
+		}
+	}
+
+	return -EINVAL;
+}
+
+static uint32_t bellboard_local_max_channels_get(const struct device *dev)
+{
+	uint32_t max_channels = 0U;
+
+	ARG_UNUSED(dev);
+
+	for (uint8_t i = 0U; i < ARRAY_SIZE(grps); i++) {
+		if (grps[i] != NULL) {
+			max_channels += NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT;
+		}
+	}
+
+	return max_channels;
+}
+
+static int bellboard_local_register_callback(const struct device *dev, uint32_t id,
+					     mbox_callback_t cb, void *user_data)
+{
+	struct mbox_bellboard_local_group *grp;
+	uint8_t grp_idx;
+	int ret;
+
+	ARG_UNUSED(dev);
+
+	ret = bellboard_local_grp_get(id, &grp_idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	grp = grps[grp_idx];
+
+	grp->cb[id] = cb;
+	grp->user_data[id] = user_data;
+
+	return 0;
+}
+
+static int bellboard_local_set_enabled(const struct device *dev, uint32_t id, bool enable)
+{
+	struct mbox_bellboard_local_group *grp;
+	uint8_t grp_idx;
+	int ret;
+
+	ARG_UNUSED(dev);
+
+	ret = bellboard_local_grp_get(id, &grp_idx);
```

Comment:
This finds only the first group to which a given event is assigned but it is technically possible that one event is assigned to several groups. It would be good to handle this possibility.

Reply:
reworked driver to support multi-irq mapping